api/health.go

```diff
@@ -95,40 +95,63 @@ func (d *HealthCheckDefinition) MarshalJSON() ([]byte, error) {
 	return json.Marshal(out)
 }
 
-func (d *HealthCheckDefinition) UnmarshalJSON(data []byte) error {
+func (t *HealthCheckDefinition) UnmarshalJSON(data []byte) (err error) {
 	type Alias HealthCheckDefinition
 	aux := &struct {
-		Interval                       string
-		Timeout                        string
-		DeregisterCriticalServiceAfter string
+		IntervalDuration                       interface{}
+		TimeoutDuration                        interface{}
+		DeregisterCriticalServiceAfterDuration interface{}
 		*Alias
 	}{
-		Alias: (*Alias)(d),
+		Alias: (*Alias)(t),
 	}
 	if err := json.Unmarshal(data, &aux); err != nil {
 		return err
 	}
 
 	// Parse the values into both the time.Duration and old ReadableDuration fields.
-	var err error
-	if aux.Interval != "" {
-		if d.IntervalDuration, err = time.ParseDuration(aux.Interval); err != nil {
-			return err
+
+	if aux.IntervalDuration == nil {
+		t.IntervalDuration = time.Duration(t.Interval)
+	} else {
+		switch v := aux.IntervalDuration.(type) {
+		case string:
+			if t.IntervalDuration, err = time.ParseDuration(v); err != nil {
+				return err
+			}
+		case float64:
+			t.IntervalDuration = time.Duration(v)
```

Comment:
Do we need to handle the floats here? Will the std json encoder ever emit floats for a `time.Duration` field?

Reply:
By default the json encoder will emit a float representation of a time.Duration, not the duration string (`100` not `"100ns"`). However, if whatever code is doing encoding is internal to consul, `json.Marshal` will use the `MarshalJSON` method above, which will emit a string representation of the duration (well, for 3 of 4 of the durations, that is). 

i added this initially for safety, and to be consistent with all the other duration decodings.

LMK what you think.

Reply:
That makes sense. I was just mistaken about the std json encoder encoding it in its string form. Apparently the Text Marshalling interface would have to be implemented by `time.Duration` for that to happen.

So nothing to see here. All is well.